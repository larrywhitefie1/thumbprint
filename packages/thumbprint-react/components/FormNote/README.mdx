---
package: '@thumbtack/thumbprint-react'
kit: form-note/index.mdx
platform: react
url: /components/form-note/react/
mdxType: componentApi
---

## Examples

### Basic form note

```jsx
<React.Fragment>
    <Label for="example-39189">Password</Label>
    <Input id="example-39189" type="password" onChange={() => {}} />
    <div className="mt1">
        <FormNote>Passwords must be at least 8 characters long.</FormNote>
    </div>
</React.Fragment>
```

### Form note describing an error

```jsx
<React.Fragment>
    <Label for="example-291929" text="Email address" hasError>
        Email address
    </Label>
    <Input id="example-291929" hasError value="example@exampl" onChange={() => {}} />
    <div className="mt1">
        <FormNote hasError>The email address you entered is incorrect.</FormNote>
    </div>
</React.Fragment>
```
