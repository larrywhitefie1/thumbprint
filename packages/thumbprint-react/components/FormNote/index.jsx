import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './index.module.scss';

const propTypes = {
    /**
     * Text within the form note.
     */
    children: PropTypes.node,
    /**
     * Renders the form note as red text.
     */
    hasError: PropTypes.bool,
};

const defaultProps = {
    children: null,
    hasError: false,
};

export default function FormNote({ hasError, children }) {
    return (
        <div className={classNames(styles.root, { [styles.rootError]: hasError })}>{children}</div>
    );
}

FormNote.propTypes = propTypes;
FormNote.defaultProps = defaultProps;
