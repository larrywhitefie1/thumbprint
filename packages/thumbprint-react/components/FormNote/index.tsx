import React from 'react';
import classNames from 'classnames';
import styles from './index.module.scss';

interface PropTypes {
    /**
     * Text within the form note.
     */
    children?: React.ReactNode;
    /**
     * Renders the form note as red text.
     */
    hasError?: boolean;
}

export default function FormNote({ hasError = false, children = null }: PropTypes): JSX.Element {
    return (
        <div className={classNames(styles.root, { [styles.rootError]: hasError })}>{children}</div>
    );
}
