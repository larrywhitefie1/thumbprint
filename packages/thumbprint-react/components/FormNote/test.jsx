import React from 'react';
import { mount } from 'enzyme';
import FormNote from './index';

test('renders form note text', () => {
    const wrapper = mount(<FormNote>Goose</FormNote>);
    expect(wrapper.text()).toBe('Goose');
    expect(wrapper).toMatchSnapshot();
});

test('renders form note with error', () => {
    const wrapper = mount(<FormNote hasError>Goose</FormNote>);
    expect(wrapper).toMatchSnapshot();
});
