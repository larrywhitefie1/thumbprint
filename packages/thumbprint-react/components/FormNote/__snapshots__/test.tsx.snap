// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders form note text 1`] = `
<FormNote>
  <div
    className="root"
  >
    Goose
  </div>
</FormNote>
`;

exports[`renders form note with error 1`] = `
<FormNote
  hasError={true}
>
  <div
    className="root rootError"
  >
    Goose
  </div>
</FormNote>
`;
